### PR TITLE
Add rerun_invocation method to InvocationClient

### PR DESCRIPTION
**Pull request by simonbray**

### CODE REVIEW

File: bioblend/galaxy/invocations/__init__.py [hunk 1]
```diff
@@ -128,6 +131,89 @@ def show_invocation(self, invocation_id):
...
+          actions
+
+        :type inputs_by: str
+        :param inputs_by: Determines how inputs are referenced. Can be
+          "step_index|step_uuid" (default), "step_index", "step_id", "step_uuid", or "name".
+
+        :type parameters_normalized: bool
+        :param parameters_normalized: Whether Galaxy should normalize ``params``
+          to ensure everything is referenced by a numeric step ID. Default is
+          ``False``, but when setting ``params`` for a subworkflow, ``True`` is
```
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:178]: ``params`` is not an argument of this method (it is of `WorkflowClient.invoke_workflow()`), so this help should be revised.

File: bioblend/galaxy/invocations/__init__.py [hunk 2]
```diff
@@ -128,6 +131,89 @@ def show_invocation(self, invocation_id):
...
+                         params_update: Optional[dict] = None, history_id: Optional[str] = None,
+                         history_name: Optional[str] = None, import_inputs_to_history: bool = False,
+                         replacement_params: Optional[dict] = None, allow_tool_state_corrections: bool = False,
+                         inputs_by: Optional[str] = None, parameters_normalized: bool = False):
+        """
+        Rerun a workflow invocation.
+
+        :type invocation_id: str
+        :param invocation_id: Encoded workflow invocation ID to be rerun
+
```
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:144]: Help for `inputs_update` and `params_update` is missing.

File: bioblend/galaxy/invocations/__init__.py [hunk 3]
```diff
@@ -128,6 +131,89 @@ def show_invocation(self, invocation_id):
...
+        :param parameters_normalized: Whether Galaxy should normalize ``params``
+          to ensure everything is referenced by a numeric step ID. Default is
+          ``False``, but when setting ``params`` for a subworkflow, ``True`` is
+          required.
+
+        :rtype: dict
+        :return: A dict describing the new workflow invocation.
+
+        .. note::
+          This method can only be used with Galaxy ``release_21.05`` or later.
```
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:185]: Why? Doesn't it use the same API endpoint of `WorkflowClient.invoke_workflow()` ?

Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py:196]: Unfortunately this PR is required: https://github.com/galaxyproject/galaxy/pull/11871

**Comment by simonbray** [comment]
Thanks for spotting those mistakes @nsoranzo.

File: bioblend/galaxy/invocations/__init__.py (hunk 3 above)
Comment (nsoranzo) [reply bioblend/galaxy/invocations/__init__.py:196]: I see, thanks. If you need it for an earlier Galaxy release, I think the changes to `lib/galaxy/model/__init__.py` could be backported as a bugfix.

**Review by nsoranzo — APPROVED** [review]

File: bioblend/galaxy/invocations/__init__.py (hunk 3 above)
Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py:196]: I don't need it personally, so I will go ahead and merge.

File: bioblend/galaxy/invocations/__init__.py [hunk 4]
```diff
@@ -128,6 +131,100 @@ def show_invocation(self, invocation_id):
...
+          ``True`` is required.
+
+        :rtype: dict
+        :return: A dict describing the new workflow invocation.
+
+        .. note::
+          This method can only be used with Galaxy ``release_21.05`` or later.
+        """
+        invocation_details = self.show_invocation(invocation_id)
+        workflow_id = invocation_details['stored_workflow_id']
```
Comment (mvdbeek) [inline bioblend/galaxy/invocations/__init__.py:199]: This shouldn't be part of the public API and I thought we agreed on that in https://github.com/galaxyproject/galaxy/pull/11881#discussion_r618314707

Comment (mvdbeek) [reply bioblend/galaxy/invocations/__init__.py:199]: And I see you've changed that in a later commit :)

**Pull request merged**